app/helpers/proxmox_vm_config_helper.rb

```diff
@@ -126,6 +126,11 @@ def parsed_typed_config(args, type)
   def parse_typed_memory(args, type)
     ForemanFogProxmox::HashCollection.remove_empty_values(args)
     logger.debug("parse_typed_memory(#{type}): args=#{args}")
+    # Convert balloon value from bytes to expected MiB if its not already
+    if args.key?('balloon')
+      mb = (args['balloon'].to_i / 1024**2).to_s
+      args['balloon'] = (mb == '0') ? args['balloon'] : mb
```

Comment:
I am not sure if this is a good idea, because in the case where `balloon` is `nil` or something < 1024^2, we should not regard it as a valid value and maybe not send it at all or fail.

If someone says for instance: 'I want to set 500KB'. In this case we would set the value to ~500,000, which IMHO will be interpreted as 500GB.

If we have to have this line, please make sure that a warning is logged.

Reply:
On GUI side, user needs to add the value in MB only which the GUI already gives hint of. The issue was even if user doesn't provide any value, when the host is powered on, on backend side the balloon value was set to maximum value of memory and somehow it gets converted to bytes and set on foreman side and thus throwing error " balloon value cannot be greater than memory" (it was comparing bytes to MB).

Reply:
OK, so the problem is we cannot be sure if the value is in bytes or megabytes.

Is there a chance that -in the future- we could know for sure which unit the value is in?

Anyway, for now I guess it is good enough :smiley: